S3/consumers.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+
+from channels.generic.websocket import AsyncJsonWebsocketConsumer
+
+from S3.models import Hash, S3
+from analytics.models import CapturedData
+from analytics.serializers import GetCapturedDataSerializer
+
+
+class AdPageConsumer(AsyncJsonWebsocketConsumer):
+
+    async def connect(self):
+        if self.scope.get('url_route', None) and self.scope['url_route'].get('kwargs', None):
+            hashed_value = self.scope["url_route"]["kwargs"]['hashed_value']
+
+            try:
+                # When only protocol is fit and hash value is valid.
+                await Hash.objects.aget(hash_value=hashed_value)
+                return await self.accept()
+            except Hash.DoesNotExist:
+                return
+        else:
+            return
+
+    async def receive_json(self, content, **kwargs):
+        if content.get('captured_data', None):
+            captured_data_id = content['captured_data']
+            try:
+                c = await CapturedData.objects.aget(id=captured_data_id)
+                s = GetCapturedDataSerializer(c)
```

Comment:
It is difficult to understand what variables `c` and `s` mean.  ` c == capturedData && s == serializer`?